### PR TITLE
Clean up residual and linearization

### LINKED ISSUE #53: Clean up residual and linearization
While writing up the Methods section of the upcoming 0D-3D-UQ paper with @richterjakob, @menon-karthik, and others, I realized that our definition of the residual and its linearization is a bit more complicated than it needs to be. With $\mathbf\alpha$ being the 0D element parameters, the residual is

$$\textbf{r}\left(\dot{\textbf{y}}, \textbf{y}, \mathbf\alpha\right) = \textbf{E}\left(\textbf{y}, \mathbf\alpha \right) \cdot \dot{\textbf{y}} + \textbf{F}\left( \textbf{y}, \mathbf\alpha \right) \cdot \textbf{y} + \textbf{c}\left(\textbf{y}, \mathbf\alpha \right),$$

and the linearization

$$\textbf{K}\left(\dot{\textbf{y}}, \textbf{y}, \mathbf\alpha \right) = \underset{\text{Term 1}}{\underbrace{\frac{\partial \textbf{E}\left(\textbf{y}, \mathbf\alpha \right)}{\partial \textbf{y}}\cdot\dot{\textbf{y}}}} + \frac{\alpha\_{m}}{\Delta t\alpha\_{f}\gamma}\textbf{E}\left(\textbf{y}, \mathbf\alpha \right) + \underset{\text{Term 2}}{\underbrace{\frac{\partial \textbf{F}\left(\textbf{y}, \mathbf\alpha \right)}{\partial \textbf{y}}\cdot\textbf{y}}} + \textbf{F}\left(\textbf{y}, \mathbf\alpha \right) + \underset{\text{Term 3}}{\underbrace{\frac{\partial \textbf{c}\left(\textbf{y}, \mathbf\alpha \right)}{\partial \textbf{y}}}}.$$

However, this adds confusion on where to put nonlinear terms ($\textbf{E}$, $\textbf{F}$, or $\textbf{c}$) and also adds three partial derivatives. Therefore, I'm proposing the following residual

$$\textbf{r}\left(\dot{\textbf{y}}, \textbf{y}, \mathbf\alpha\right) = \textbf{E}\left(\mathbf\alpha \right) \cdot \dot{\textbf{y}} + \textbf{F}\left( \mathbf\alpha \right) \cdot \textbf{y} + \textbf{c}\left(\textbf{y}, \mathbf\alpha \right).$$

Here, $\textbf{E}$ and $\textbf{F}$ are constant, and all nonlinear terms are in $\textbf{c}$. The linearzation is also simplified as 

$$\mathbf K (\mathbf y, \dot{\mathbf y}, \boldsymbol \alpha) 
= \frac{\alpha_m}{\alpha_f\gamma\Delta t}  \left( \mathbf E + \frac{\partial\mathbf c}{\partial\dot{\mathbf y}} \right) + \mathbf F + \frac{\partial\mathbf c}{\partial\mathbf y}.$$

This requires only the two partial derivatives of the nonlinear term $\textbf{c}$.

I think, in the end, the assembled matrices should look the same in the solver (but this will serve as another sanity check). This will only affect blocks with nonlinear elements, which I think are `BloodVessel`, `BloodVesselJunction`, and `ClosedLoopHeartPulmonary`. I'll make the changes to those blocks and `SparseSystem`, and update the Doxygen documentation. I'll also move any (non-redundant) content from the [ROM Simulation Guide](https://github.com/SimVascular/simvascular.github.io/edit/main/documentation/rom_simulation/0d-solver/theory/readme.md) to Doxygen, remove the documentation in [simvascular.github.io](https://github.com/SimVascular/simvascular.github.io), and link to Doxygen.

### PR DESCRIPTION
**Pull request by mrp089**
Closes #53.

Code changes:
- Cleaned up residual and linearization as described in #53 
- Add output for average number of Newton iterations per time step

Documentation:
- Adapted documentation to new formulation
- Added documentation from [simvascular.github.io](https://simvascular.github.io/documentation/rom_simulation.html#0d-solver)

Testing:
- Same results
- Same number of Newton iterations
- Same runtime (should be a tiny bit faster now)

### CODE REVIEW

**Comment by mrp089** [comment]
I also changed our LaTeX rendering to MathJax since it looks nicer. However, this got me into a rabbit hole:

- MathJax only supports [a couple of LaTeX packages](https://docs.mathjax.org/en/latest/input/tex/extensions/index.html). We use `CircuiTikZ` package for the schematics of 0D blocks, which is not supported.
- I thus compiled all schematics and saved them under `docs/png`, one for light mode and one for dark mode
- `GitHub` markdown allows switching images by toggling light/dark mode. However, this feature is (currently) not supported in `Doxygen`.

I thus defaulted to the dark version of each schematic. I think that's fine since it's still readable in light mode, and we also have only one light mode in our [current documentation](https://simvascular.github.io/documentation/rom_simulation.html#0d-solver). Maybe in a future version of `Doxygen` we can include an automatic switch.

File: docs/Doxyfile [hunk 1]
```diff
@@ -189,7 +189,7 @@ ENUM_VALUES_PER_LINE   = 1
 TREEVIEW_WIDTH         = 250
 EXT_LINKS_IN_WINDOW    = NO
 FORMULA_FONTSIZE       = 10
-USE_MATHJAX            = NO
```
Comment (menon-karthik) [inline docs/Doxyfile:192]: If someone wants to add a new block that does not already have a CircuiTikZ schematic, do we document somewhere how they can generate one?

Comment (mrp089) [reply docs/Doxyfile:192]: I'll add something to the developer guide

Comment (mrp089) [reply docs/Doxyfile:192]: @menon-karthik plz have a look!

File: docs/pages/developer_guide.md [hunk 2]
```diff
@@ -75,33 +75,53 @@ inspiration, you can look at the existing source files and how they use
...
+3. Copy `png` output from `docs/build/latex` to `docs/png`
+4. Include image with `\image html FILENAME_dark.png` in `.h`/`.cpp` Doxygen
+5. Set `USE_MATHJAX = YES` in `docs/Doxyfile`
+6. Locally build Doxygen again
+7. Check if LaTeX and schematic look nice
+8. Commit your changes and new `png`
+
 
+### Build
 The documentation is automatically build in the GitHub CI/CD and published
```
Comment (menon-karthik) [inline docs/pages/developer_guide.md:118]: Last comment - "build" should be "built"

**Review by menon-karthik — APPROVED** [review]

File: docs/Doxyfile (hunk 1 above)
Comment (menon-karthik) [reply docs/Doxyfile:192]: Thanks!

**Pull request merged**